src/zcash/JoinSplit.cpp

```diff
@@ -266,8 +267,14 @@ class JoinSplitCircuit : public JoinSplit<NumInputs, NumOutputs> {
 
         std::stringstream ss;
         ss << proof;
+        std::string serialized_proof = ss.str();
 
-        return ss.str();
+        boost::array<unsigned char, ZKSNARK_PROOF_SIZE> result_proof;
+        //std::cout << "proof size in bytes when serialized: " << serialized_proof.size() << std::endl;
+        assert(serialized_proof.size() == ZKSNARK_PROOF_SIZE);
+        memcpy(&result_proof[0], &serialized_proof[0], ZKSNARK_PROOF_SIZE);
```

Comment:
Never mind, std::string is a basic_string which says "The elements of a basic_string are stored contiguously, that is, for a basic_string s, &*(s.begin() + n) == &*s.begin() + n for any n in [0, s.size()), or, equivalently, a pointer to s[0] can be passed to functions that expect a pointer to the first element of a CharT[] array."  http://en.cppreference.com/w/cpp/string/basic_string